src/lib.rs

```diff
@@ -1,6 +1,123 @@
-//! This crate provides a `Watch` that launch a given command, re-launching this
-//! command when changes are detected in your source code.
-
+//! This crate provides a [`Watch`] that launch a given command, re-launching
+//! the command when changes are detected in your source code.
+//!
+//! This [`Watch`] is based on the intended to be used on projects that rely on
+//! the [xtask concept](https://github.com/matklad/cargo-xtask/) and implement
+//! [`clap::Parser`] to be added easily to an existing CLI implementation.
```

Comment:
```suggestion
//! This [`Watch`] struct is intended to be used with
//! the [xtask concept](https://github.com/matklad/cargo-xtask/) and implements
//! [`clap::Parser`] so it can easily be used in your xtask crate. See [clap's `flatten`](link-to-flatten) to see how to extend it.
```